packages/ckeditor5-ui/src/bindings/clickoutsidehandler.js

```diff
@@ -25,13 +25,14 @@
  * @param {Function} options.callback An action executed by the handler.
  */
 export default function clickOutsideHandler( { emitter, activator, callback, contextElements } ) {
-	emitter.listenTo( document, 'mousedown', ( evt, { target } ) => {
+	emitter.listenTo( document, 'mousedown', ( evt, domEvt ) => {
 		if ( !activator() ) {
 			return;
 		}
 
 		for ( const contextElement of contextElements ) {
-			if ( contextElement.contains( target ) ) {
+			if ( contextElement.contains( domEvt.target ) ||
+						( 'composedPath' in domEvt && domEvt.composedPath().includes( contextElement ) ) ) {
```

Comment:
Maybe add a comment on why you're checking for the existence of `composedPath`. I'm not sure how this project tracks browser support, but they may want to remove then when all supported browsers have native shadow DOM.

Reply:
A comment would great! We don't unfortunately track such things in a more organized way yet, so a comment is the best we can do for now.